service/amount_out.go

```diff
@@ -95,6 +99,10 @@ func getOutAmountInBToken(ctx context.Context, it models.OrderIter, inAmountA de
 	var frags []models.OrderFrag
 	for it.HasNext() && inAmountA.IsPositive() {
 		order = it.Next(ctx)
+		if order == nil {
```

Comment:
Can you update the Error log to Warn then as this is not an unexpected error, and change the error? You could do models.ErrInsufficientLiquidity for example?